src/servant/tt_002/faerie_ability_data.c

```diff
@@ -0,0 +1,127 @@
+    {0, 0}, {4, 1}, {4, 1}, {4, 1}, {4, 1}, {4, 1}, {4, 1}, {4, 1},
+    {4, 1}, {4, 1}, {4, 1}, {4, 1}, {4, 1}, {4, 1}, {4, 1}, {4, 1},
+};
+
+FaerieAbilityStats g_FaerieAbilityStats[] = {
+    {.timer = 0x1E, // lvl 0-9
+     .unk1 = 0xFF,
+     .hammerChance = 0x20,
+     .uncurseChance = 0x10,
+     .antivenomChance = 0x10,
+     .resistChance = 0x10,
+     .healChance = 0xC0},
+    {.timer = 0x14, // lvl 10-19
+     .unk1 = 0xFF,
+     .hammerChance = 0x20,
+     .uncurseChance = 0x10,
+     .antivenomChance = 0x10,
+     .resistChance = 0x20,
+     .healChance = 0xC0},
+    {.timer = 0x0F, // lvl 20-29
+     .unk1 = 0xFF,
+     .hammerChance = 0x40,
+     .uncurseChance = 0x20,
+     .antivenomChance = 0x20,
+     .resistChance = 0x20,
+     .healChance = 0xD0},
+    {.timer = 0x0A, // lvl 30-39
+     .unk1 = 0xFF,
+     .hammerChance = 0x40,
+     .uncurseChance = 0x20,
+     .antivenomChance = 0x20,
+     .resistChance = 0x40,
+     .healChance = 0xD0},
+    {.timer = 0x0A, // lvl 40-49
+     .unk1 = 0xFF,
+     .hammerChance = 0x60,
+     .uncurseChance = 0x20,
+     .antivenomChance = 0x30,
+     .resistChance = 0x40,
+     .healChance = 0xE0},
+    {.timer = 0x05, // lvl 50-59
+     .unk1 = 0xFF,
+     .hammerChance = 0x60,
+     .uncurseChance = 0x40,
+     .antivenomChance = 0x30,
+     .resistChance = 0x40,
+     .healChance = 0xE0},
+    {.timer = 0x03, // lvl 60-69
+     .unk1 = 0xFF,
+     .hammerChance = 0x80,
+     .uncurseChance = 0x40,
+     .antivenomChance = 0x40,
+     .resistChance = 0x60,
+     .healChance = 0xE0},
+    {.timer = 0x02, // lvl 70-79
+     .unk1 = 0xFF,
+     .hammerChance = 0xA0,
+     .uncurseChance = 0x60,
+     .antivenomChance = 0x50,
+     .resistChance = 0x60,
+     .healChance = 0xFF},
+    {.timer = 0x01, // lvl 80-89
+     .unk1 = 0xFF,
+     .hammerChance = 0xC0,
+     .uncurseChance = 0x60,
+     .antivenomChance = 0x60,
+     .resistChance = 0x80,
+     .healChance = 0xFF},
+    {.timer = 0x00, // lvl 90-99
+     .unk1 = 0xFF,
+     .hammerChance = 0xC0,
+     .uncurseChance = 0x60,
+     .antivenomChance = 0x60,
+     .resistChance = 0x80,
+     .healChance = 0xFF},
+};
+
+// This may be able to be broken into a 4 short struct
+// unknown param for D_80139828, ItemId, FactoryParam, EventParam
+s16 g_ResistItemsParamMap[] = {
+    0x0005, ITEM_RESIST_FIRE,    0x0009, 0x0005,
+    0x0007, ITEM_RESIST_THUNDER, 0x000A, 0x0006,
+    0x0006, ITEM_RESIST_ICE,     0x000B, 0x0007,
+    0x0009, ITEM_RESIST_HOLY,    0x000E, 0x0008,
+    0x000B, ITEM_RESIST_DARK,    0x000F, 0x0009,
+    0x000A, ITEM_RESIST_STONE,   0x000C, 0x000A,
+};
+
+// itemId, FACTORY() param
+// check if can be made struct, but PSP may be a problem.
+s16 g_PotionItemsParamMap[] = {
+    ITEM_POTION,
+    0x0010,
+    ITEM_HIGH_POTION,
+    0x0011,
+};
+
+// This is an array of type Unkstruct_801724CC, but due to packing,
+// the raw data needs to be a u16 array.
+u16 D_us_801724CC[] = {
```

Comment:
The problem is too much or too little padding? You can maybe manipulate it https://stackoverflow.com/questions/11770451/what-is-the-meaning-of-attribute-packed-aligned4

Reply:
I tried both #pragma pack(push,1) and then popping it after as well as using __attribute__((packed)) and some various variants of those to no avail. If I built this an an array of the correct type, we always got padded 2 extra bytes. It seems like the compiler just ignored all the packing commands. 

I did try for a few hours before giving in and doing it like this.  If somebody else wants to take a crack at it, I’d be happy to update if they figure it out.